### ISSUE TITLE
Update control items and text input API to allow max-width management

### ISSUE THREAD

**Issue by paulinea** [Orange-OpenSource/ouds-flutter#533]
A user should be able to choose if a control item or a text input must be displayed to fill its container in width, or using a max-width to constrain its width on large screens. It should maybe be applied to other components on Flutter like password input.

Add a `constrainedMaxWidth` boolean parameter to `OudsSwitchItem`, `OudsCheckboxItem`, `OudsRadioButtonItem`, `OudsTextInput`, `OudsPinCodeInput`, `OudsPasswordInput` and `OudsPhoneNumberInput`:
- **true** Apply the max-width token to limit component's width in large area.
- **false** The component occupies the entire width of its container.

**Comment by AhmedAmineZr** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: DesignToolbox (alpha) (593)
Version: 0.7.0
Build type: Alpha
Build details (GitHub): 533
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 @Tayebsed93 (dev team)
cc @jerome-regnier (design team)

**Comment by AhmedAmineZr** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@ AhmedAmineZr) to get access.

Display name: Design Toolbox Flutter (593)
Version: 0.7.0
Build type: Alpha
Build details (GitHub): 533
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by jerome-regnier** [comment]
Review fonctionnelle Design sur l'App de test flutter sur iOS et Android.

Ok Design

**Comment by B3nz01d** [comment]
The order in witch all this new property is added is not always set the same way...
For the controls is it possible to set this order:
* icon
* divider
* outlined
* enabled
* read only
* error
* reversed
* constrained max width

For the inputs is it possible to set the following order:
* leading icon
* country selector
* prefix
* hidden password
* trailing icon
* loader
* helper text
* length
* enabled
* read only
* error
* outlined
* rounded corner
* constrained max width

**Comment by AhmedAmineZr** [comment]
> The order in witch all this new property is added is not always set the same way... For the controls is it possible to set this order:
> 
> * icon
> * divider
> * outlined
> * enabled
> * read only
> * error
> * reversed
> * constrained max width
> 
> For the inputs is it possible to set the following order:
> 
> * leading icon
> * country selector
> * prefix
> * hidden password
> * trailing icon
> * loader
> * helper text
> * length
> * enabled
> * read only
> * error
> * outlined
> * rounded corner
> * constrained max width

i will change the position of `constrained max width` in the bottom of components and the the change of order will be done into separate ticket [#540](https://github.com/Orange-OpenSource/ouds-flutter/issues/540)

**Comment by AhmedAmineZr** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: DesignToolbox (alpha) (597)
Version: 0.7.0
Build type: Alpha
Build details (GitHub): 533
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 @Tayebsed93 (dev team)
cc @jerome-regnier (design team)

**Comment by AhmedAmineZr** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@ AhmedAmineZr) to get access.

Display name: Design Toolbox Flutter (597)
Version: 0.7.0
Build type: Alpha
Build details (GitHub): 533
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
